src/Framework/Microsoft.AspNetCore.App.props

```diff
@@ -152,7 +79,26 @@
       This will help us ensure servicing builds carry the latest versions of dependencies, even if we aren't rebuilding
       the original direct dependency that pulled in the transitive reference.
     -->
-    <ExternalDependency Include="@(_TransitiveExternalDependency)" />
+    <ExternalAspNetCoreAppReference Include="@(_TransitiveExternalAspNetCoreAppReference)" />
+  </ItemGroup>
+
+  <!--
+    These compilation references are necessary to compile ns2.0 assemblies in the shared framework, but the runtime assemblies are part of netcoreapp3.0.
+    The preferred solution to reference problems with shared framework assemblies is to reference netcoreapp3.0 and not use System.* references,
+    but in some edge cases, this is not reasonable.
+  -->
+  <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
+    <_FrameworkProvidedReference Include="System.Buffers" />
```

Comment:
I don't completely understand this. Is this just to avoid producing errors about referencing `System.Buffers` ?

Reply:
Can we flow this information via ExternalAspNetCoreAppReference somehow?

Reply:
Correct. A few signalR assemblies target netstandard2.0 and need the System.Buffers reference, but this dependency is knocked out in netcoreapp3.0. I wanted to provide a way for us to require explicit action to put new assemblies into the shared framework.

Reply:
I updated the name and comments to clarify that this is a "compilation only reference". This list should be fairly stable since netstandard2.0 is set in stone. The only thing which might change is what is in netcoreapp3.0.